File: src/ptit-rental-car-api-resource/src/ApplicationCore/Entities/VehicleHandover.cs
using NGOT.Common.Enums;
using NGOT.Common.Models;

namespace NGOT.ApplicationCore.Entities;

public class VehicleHandover : Entity<Guid>
{
    public DateTime HandoverDate { get; set; }
    public string HandoverBy { get; set; } = null!;
    public HandoverTypeEnum HandoverType { get; set; }
    public string RentalContractId { get; set; } = null!;
    public RentalContract RentalContract { get; set; } = null!;
    public virtual List<CheckListItem> CheckListItems { get; set; } = new();
}

File: src/ptit-rental-car-ui/libs/data-models/src/lib/vehicle-handover/vehicle-handover.ts
export enum HandoverTypeEnum {
  Delivery= "Delivery",
  Return = "Return",
}

export const HandoverTypeEnumDescriptions: Record<HandoverTypeEnum, string> = {
  [HandoverTypeEnum.Delivery]: "Giao xe",
  [HandoverTypeEnum.Return]: "Nhận xe",
}

export enum CheckListItemStatusEnum {
  Ok = "Ok",
  Repair = "Repair",
}

export const CheckListItemStatusEnumDescriptions: Record<CheckListItemStatusEnum, string> = {
  [CheckListItemStatusEnum.Ok]: "Đạt",
  [CheckListItemStatusEnum.Repair]: "Sửa chữa",
}

export interface CheckListItem {
  id: string;
  name: string;
  status: CheckListItemStatusEnum;
  comment: string;
  vehicleHandoverId: string;
}

export interface VehicleHandover {
  id: string;
  handoverDate: Date;
  handoverBy: string;
  handoverType: HandoverTypeEnum;
  rentalContractId: string;
  checkListItems: CheckListItem[];
}
